### PR TITLE
[usecase-linux][XWALK-286] Add Web notification in Linux

### PR DESCRIPTION
**Pull request by xinliux**
- Add Web Notification tests for Linux

Impacted tests(approved): new 1, update 0, delete 0
Unit test platform: [Linux]
Unit test result summary: pass 0, fail 1, block 0

Failure reason: Click "Show Notification" button, Web Notification constructor function throw error message "Failed to send message to remove remote en dpoint"

### CODE REVIEW

File: usecase/usecase-wrt-linux-tests/samples/WebNotification/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+## Usecase Design
+
+This sample demonstrate Web Notification feature basic functionalities, include:
+
+* Notification serviceWorker register, showNotification and notificationclick
```
Comment (zqzhang) [inline usecase/usecase-wrt-linux-tests/samples/WebNotification/README.md]: At https://crosswalk-project.org/jira/browse/XWALK-3654, Crosswalk makes a workaround of the Web Notifications implementation. 'Desktop' notification doesn't implement the latest push and service worker based notification. It should be based on W3C spec, but not the WHATWG spec. Please double check it.

**Comment by zqzhang** [comment]
> Add Notification serviceWorker register, showNotification and notificationclick

The code here has nothing to do with this commit message.

Another, does it still fail? What is the fail info?

File: usecase/usecase-wrt-linux-tests/samples/WebNotification/index.html [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+<script src="../../js/tests.js"></script>
+<script src="js/main.js"></script>
+
+<div id="header">
+  <h3 id="main_page_title"></h3>
+</div>
+<div class="content">
+  <h4>This sample use the device's native notification mechanism to display messages to the user.</h4>
+  <div>
+    <button onclick="javascript:showNotification()" id="showNotification" type="button" class="tagBtn btn btn-default btn-lg btn-block">Show Notification</button>
```
Comment (zqzhang) [inline usecase/usecase-wrt-linux-tests/samples/WebNotification/index.html]: Please split JS and CSS code from HTML, thanks.

File: usecase/usecase-wrt-linux-tests/samples/WebNotification/index.html [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  </div>
+  <div style="margin-top: 5px">
+    <button onclick="javascript:closeNotification()" id="closeNotification" type="button" class="tagBtn btn btn-default btn-lg btn-block">Close Notification</button>
```
Comment (zqzhang) [inline usecase/usecase-wrt-linux-tests/samples/WebNotification/index.html]: Please split JS and CSS code from HTML, thanks.

File: usecase/usecase-wrt-linux-tests/samples/WebNotification/index.html [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+Authors:
+        Liu.Xin <xinx.liu@intel.com>
+
+-->
+
+<meta charset="utf-8" />
+<meta name="viewport" content="initial-scale=1.0, maximum-scale=1.0, width=device-width" />
+<link rel="stylesheet" type="text/css" href="../../css/bootstrap.css">
```
Comment (zqzhang) [inline usecase/usecase-wrt-linux-tests/samples/WebNotification/index.html]: Remove `type="text/css"`

File: usecase/usecase-wrt-linux-tests/samples/WebNotification/index.html [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+<link rel="stylesheet" type="text/css" href="../../css/main.css">
```
Comment (zqzhang) [inline usecase/usecase-wrt-linux-tests/samples/WebNotification/index.html]: ditto

File: usecase/usecase-wrt-linux-tests/samples/WebNotification/index.html [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+<div id="header">
+  <h3 id="main_page_title"></h3>
+</div>
+<div class="content">
+  <h4>This sample use the device's native notification mechanism to display messages to the user.</h4>
+  <div>
+    <button onclick="javascript:showNotification()" id="showNotification" type="button" class="tagBtn btn btn-default btn-lg btn-block">Show Notification</button>
+  </div>
+  <div style="margin-top: 5px">
```
Comment (zqzhang) [inline usecase/usecase-wrt-linux-tests/samples/WebNotification/index.html]: Don't mix the style within element

File: usecase/usecase-wrt-linux-tests/samples/WebNotification/js/main.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+var notification;
+var button_Flag = false;
+
+jQuery(document).ready(function() {
+  $("#showNotification").attr('disabled', button_Flag);
+  $("#closeNotification").attr('disabled', !button_Flag);
+});
+
+function showNotification() {
+  if(Notification.permission === "granted") {
```
Comment (zqzhang) [inline usecase/usecase-wrt-linux-tests/samples/WebNotification/js/main.js]: Put a space before `(`, please check all your test code,

File: usecase/usecase-wrt-linux-tests/steps/WebNotification/step.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+               + '<li>'
+                 + 'Click the "Show Notification" button.'
+               + '</li>'
+               + '<li>'
+                 + 'Click the "Close Notification" button.'
+               + '</li>'
+             + '</ol>'
+           + '</p>'
+           + '<p>Expected Result: </p>'
+           + '<p>Test passes if show a notification after step 1, and then this notification close after step 2.</p>'
```
Comment (zqzhang) [inline usecase/usecase-wrt-linux-tests/steps/WebNotification/step.js]: Make 2 expects results against the 2 steps respectively.

**Comment by xinliux** [comment]
@zqzhang ,  These PR  update again , Please review , thanks

**Comment by zqzhang** [comment]
Still cannot merge it. Tests.xml has been changed, I guess.

**Pull request merged**